pkg/server/bundle/datastore/wrapper.go

```diff
@@ -0,0 +1,45 @@
+package datastore
+
+import (
+	"context"
+	"crypto"
+	"time"
+
+	"github.com/spiffe/spire/pkg/server/bundle/pubmanager"
+	"github.com/spiffe/spire/pkg/server/datastore"
+	"github.com/spiffe/spire/proto/spire/common"
+)
+
+// WithBundlePublisher wraps a datastore interface and provides updates to
+// bundle publishers in operations that modify the local bundle.
+func WithBundlePublisher(ds datastore.DataStore, pubManager pubmanager.PubManager) datastore.DataStore {
+	return datastoreWrapper{
+		DataStore:  ds,
+		pubmanager: pubManager,
+	}
+}
+
+type datastoreWrapper struct {
+	datastore.DataStore
+	pubmanager pubmanager.PubManager
+}
+
+func (w datastoreWrapper) AppendBundle(ctx context.Context, bundle *common.Bundle) (_ *common.Bundle, err error) {
+	defer w.pubmanager.BundleUpdated()
```

Comment:
Is it correct to always call `BundleUpdated` even though it could be the case that the `AppendBundle`(and `PruneBundle`) didn't change the bundle?

Reply:
I think that AppendBundle and PruneBundle in this wrapper will always be called in the context of operations of this server (always with the trust domain of the server) so I believe that it should always result in an updated trust bundle. In any case, bundle publisher plugins will periodically be called to publish the bundle (most of the times with the same bundle as the previous call) and it's up to the plugin to choose the best logic to decide if it should be pushed to the remote store or not, deal with retries, etc.

Reply:
Ok, thanks for the clarification.